### PR TITLE
Add support for Azure DevOps URL scheme

### LINKED ISSUE #1462: Add support for Azure DevOps url scheme
Would be nice to have support for URL links when hosting code on Azure DevOps

They are of the form:
`https://dev.azure.com/ORG/TEAM/_git/REPO?path={path}&version={commit}&line={line_low}&lineEnd={line_high}&lineStartColumn=1&lineEndColumn=1&lineStyle=plain`

The `{line}` one is easy: `https://dev.azure.com/ORG/TEAM/_git/REPO?path={path}&version={commit}{line}&lineStartColumn=1&lineEndColumn=1&lineStyle=plain` with `LineRangeFormatting("&line=",  "&lineEnd=")`

The commit is a bit harder... ADO annoyingly prefixes the hash with "GC" or a branch name with "GB" - so this would need a special check.

I can take a crack at doing a PR for this if that would be helpful?

### PR DESCRIPTION
**Pull request by friggog**
Resolves #1462

### CODE REVIEW

File: src/Documenter.jl [hunk 1]
```diff
@@ -160,6 +160,12 @@ For example if you are using GitLab.com, you could use
 makedocs(repo = \"https://gitlab.com/user/project/blob/{commit}{path}#{line}\")
 ```
 
+or for AzureDevOps:
+
+```julia
+makedocs(repo = \"https://dev.azure.com/org/project/_git/repo?path={path}&version={commit}{line}&lineStartColumn=1&lineEndColumn=1\")
+```
```
Comment (mortenpi) [inline src/Documenter.jl:167]: Perhaps we could merge this into the previous code block, and use comments to indicate which hosting service it applies to? Just to keep the docstring a bit shorter.

**Review by mortenpi — COMMENTED** [review]
Other than the one docstring comment, LGTM!

File: src/Writers/HTMLWriter.jl [hunk 2]
```diff
@@ -438,7 +438,7 @@ module RD
 
     const requirejs_cdn = "https://cdnjs.cloudflare.com/ajax/libs/require.js/2.3.6/require.min.js"
     const google_fonts = "https://fonts.googleapis.com/css?family=Lato|Roboto+Mono"
-    const fontawesome_version = "5.11.2"
+    const fontawesome_version = "5.15.0"
```
[the remark this answers is not in the document]
Comment (friggog) [reply src/Writers/HTMLWriter.jl:441]: Yeah, it doesn't seem to cause any problems with other icons that I can tell

File: src/Documenter.jl (hunk 1 above)
Comment (friggog) [reply src/Documenter.jl:167]: Yeah good idea

**Comment by friggog** [comment]
@mortenpi is there any reason to use font awesome over inline SVG icons? 

From what I can see (though I'm sure there are more) the only icons used are settings cog, hamburger (can be replaced with ☰ anyway), branded repo logos and file logo for unknown repo - it seems a bit heavy to include FontAwesome and FontAwesomeBrands just for these

Happy to do a separate PR to move to inline SVGs if you think this is a good idea?

File: src/Documenter.jl [hunk 3]
```diff
@@ -154,10 +154,12 @@ value of the generated link:
   - `{path}` Path to the file in the repository
   - `{line}` Line (or range of lines) in the source file
 
-For example if you are using GitLab.com, you could use
+BitBucket, GitLab and Azure DevOps are supported along with GitHub, for example:
 
 ```julia
-makedocs(repo = \"https://gitlab.com/user/project/blob/{commit}{path}#{line}\")
+makedocs(repo = \"https://gitlab.com/user/project/blob/{commit}{path}#{line}\") # GitLab
+makedocs(repo = \"https://dev.azure.com/org/project/_git/repo?path={path}&version={commit}{line}&lineStartColumn=1&lineEndColumn=1\") # ADO
```
Comment (mortenpi) [inline src/Documenter.jl:161]:
```suggestion
makedocs(repo = \"https://dev.azure.com/org/project/_git/repo?path={path}&version={commit}{line}&lineStartColumn=1&lineEndColumn=1\") # Azure DevOps
```

**Comment by mortenpi** [comment]
LGTM, will merge shortly. Thanks a lot @friggog!

Regarding the use of SVG icons: I'd say FA is just the simplest option that gives a set of consistently styles icons. No objections to using an SVG icon if, for example, FA doesn't have an icon available.

It doesn't look like FA actually adds that much to the page load time though, so I am not sure it's worth complicating things with our own icons. We'd have to deploy the small icons on every build ourselves then. And there's also the IP side of things: where would you "source" the icons from?

**Comment by friggog** [comment]
FontAwesome has the SVGs [directly available](https://github.com/FortAwesome/Font-Awesome/tree/master/svgs) and free to use as far as I can tell. Also [simple-icons](https://github.com/simple-icons/simple-icons). I'll open a separate PR with some changes in and you can see what you think.

**Pull request closed without merging**